### PR TITLE
feat: add debug_traceTransaction rpc

### PR DESCRIPTION
**Pull request by colinlyguo**

### CODE REVIEW

File: client/client_eth.go [hunk 1]
```diff
@@ -298,3 +298,9 @@ func (c *RpcEthClient) FeeHistory(blockCount rpc.DecimalOrHex, lastBlock types.B
 	err = c.CallContext(context.Background(), &val, "eth_feeHistory", blockCount, lastBlock, rewardPercentiles)
 	return
 }
+
+// TraceTransaction returns the block trace by block number.
```
Comment (Thegaram) [inline client/client_eth.go:302]: It would be more consistent if we put this into a new file `client_debug.go`

File: types/types.go [hunk 2]
```diff
@@ -320,3 +320,21 @@ type FeeHistoryResult struct {
 	BaseFee      []*hexutil.Big   `json:"baseFeePerGas,omitempty"`
 	GasUsedRatio []float64        `json:"gasUsedRatio"`
 }
+
+type TraceTransactionResult struct {
```
Comment (Thegaram) [inline types/types.go:324]: It seems that some fields (`Failed`) are missing and some types (`Depth`, `ReturnValue`) are slightly different from geth: https://github.com/ethereum/go-ethereum/blob/master/eth/tracers/logger/logger.go#L406

Is it possible to use geth's `ExecutionResult` directly? Or is the l2geth return type different?

File: client/client_eth.go (hunk 1 above)
Comment (colinlyguo) [reply client/client_eth.go:302]: yep.

File: types/types.go (hunk 2 above)
Comment (colinlyguo) [reply types/types.go:324]: Thanks. It is possible to use `ExecutionResult`. I didn't know this before.

**Pull request closed without merging**